### DISCUSSION TITLE
How do I redirect from /<not "dashboard"> to /dashboard in next.config.mjs?

### DISCUSSION THREAD

**Discussion by Vrezerino** [vercel/next.js#64156] [App Router]
Regex starting with "?" is apparently not allowed in source.
Alternatively I wouldn't redirect, but instead I'd want to render not-found.tsx within the dashboard layout. How should I go about doing that?

**Comment by SouthLink** [comment]
Create middleware.ts under the root directory :

```
import { NextResponse } from 'next/server'
import type { NextRequest } from 'next/server'

export const config = {
  matcher: [
    // Matches internal resources of next.js
    '/((?!api|_next/static|_next/image|favicon.ico).*)',
  ],
}
 
export function middleware(request: NextRequest) {

  if (/\/(?!dashboard)/.test(request.nextUrl.pathname)) {
    return NextResponse.rewrite(new URL('/dashboard', request.url))
  }
  
  // pass the header to the layout
  return NextResponse.next();
}
```

**Reply by Vrezerino** [reply]
Nice. I don't know why I wasn't thinking that since I already had written middleware.ts for request body validation.
With redirect() though instead of rewrite(), the destination URL will update on the client side.
